pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1336,6 +1336,9 @@ type ArtGCStatus struct {
 
 	// if this is true, we already checked to see if we need to do it and we don't
 	NotSpecified bool `json:"notSpecified,omitempty" protobuf:"varint,3,opt,name=notSpecified"`
+
+	// Have task results been completed? (mapped by Pod name) used to prevent premature garbage collection of artifacts.
+	TaskResultsCompleted map[string]bool `json:"taskResultsCompleted,omitempty" protobuf:"bytes,4,opt,name=taskResultsCompleted"`
```

Comment:
can we move it to outside of the ArtGCStatus, since it's relevant outside of ArtifactGC?

Reply:
Sure, do you have a preference?

Reply:
Maybe in `Outputs` or top level?

Reply:
Hmm, let's go with top level?